src/engine/SpatialJoin.cpp

```diff
@@ -213,10 +213,13 @@ size_t SpatialJoin::getResultWidth() const {
 // ____________________________________________________________________________
 size_t SpatialJoin::getCostEstimate() {
   if (childLeft_ && childRight_) {
-    size_t inputEstimate =
-        childLeft_->getSizeEstimate() * childRight_->getSizeEstimate();
+    auto n = childLeft_->getSizeEstimate();
+    auto m = childRight_->getSizeEstimate();
+
+    size_t spatialJoinCostEst = 1;
+
     if (config_.algo_ == SpatialJoinAlgorithm::BASELINE) {
-      return inputEstimate * inputEstimate;
+      spatialJoinCostEst = n * m;
     } else {
       AD_CORRECTNESS_CHECK(
```

Comment:
This whole function can be simplified about controlFlowwiseÖ

```
if (any child missing) {
  return 1;
} // from here on not nested
// Compute the "actual cost"' via an immediately invoked lambda
size_t spatialJoinCostEst = [&/*capture what is needed/*]() {
  // compute m and n
if (BASELINE) { return n * m ; }
CHECK(S2 or BOOST)
return m log m + n log n;
}();
return spatialJoinCostEst + bothSizes;

```
With this you have less nesting and get rid of the dummy initialization of the estimate value.